implementation/common/src/main/java/io/smallrye/jwt/algorithm/SignatureAlgorithm.java

```diff
@@ -31,6 +33,19 @@ public String getAlgorithm() {
     }
 
     public static SignatureAlgorithm fromAlgorithm(String algorithmName) {
-        return SignatureAlgorithm.valueOf(algorithmName);
+        try {
+            return SignatureAlgorithm.valueOf(algorithmName.toUpperCase());
```

Comment:
This gives more opportunities for library users to write whatever case they prefer. OTOH, doing
```java
if (EDDSA.name().equalsIgnoreCase(algorithmName)) {
    return EDDSA;
}
return SignatureAlgorithm.valueOf(algorithmName);
```
instead would introduce compatibility-breaking change.

Reply:
@0rzech Thanks for another PR, can you clarify please what a compatibility-breaking change you have in mind ?

I was thinking about a tiny optimization instead:
```
Sttring algorithmNameUpperCase = algorithmName.toUpperCase();
if (EDDSA.name().equals(algorithmNameUpperCase)) {
    return EDDSA;
}
return SignatureAlgorithm.valueOf(algorithmName);
```

but I'm not sure where is the breaking change ?
 

As I said before, I don't think we need to support cases with random low and upper case characters in the algorithm names, `rS256` or `Rs256`, and another set of options of `EdDSA`.

Reply:
Never mind, your code looks good,  back from pto, so a bit jet lagged

Reply:
Yeah, I ignored the NPE, because the catch clause will catch it anyway. All in all, `null` algorithm name is just another variant of invalid name. ;)